c7n/commands.py

```diff
@@ -194,6 +196,8 @@ def validate(options):
     used_policy_names = set()
     structure = StructureParser()
     errors = []
+    found_deprecations = False
+    footnotes = deprecated.Footnotes()
```

Comment:
what's the purpose of footnotes, it seems like a pass through to report, why does the caller have to instantiate?

Reply:
The purpose of the footnotes is to aggregate deprecation information across multiple reports. Deprecation reports are created per-policy, whereas the footnotes are reported for all policies in all files across the validation run. Hence why they are instantiated outside.